src/calculate-deal.mo

```diff
@@ -0,0 +1,79 @@
+import Iter "mo:base/Iter";
+import Nat "mo:base/Nat";
+
+module {
+
+  public func calculateDeal<Order>(
+    asks : Iter.Iter<Order>,
+    bids : Iter.Iter<Order>,
+    getVolume : (Order) -> Nat,
+    getPrice : (Order) -> Float,
```

Comment:
Can we simplify this so that `Order` is fixed as `(Float, Nat)` and then we don't need `getVolume` and `getPrice` functions?

It should be easy for the iterator to extract price and volume from Order, whatever Order is, and to return that pair.

Reply:
done